### ISSUE TITLE
Delete account failed, claims still tickets assigned

### ISSUE THREAD

**Issue by Pummelchen** [polonel/trudesk#171]
**Is this a BUG REPORT or FEATURE REQUEST?**:

- [X] BUG 
- [ ] FEATURE

**What happened**:
I try to delete a user account, but TruDesk claims "account is linked to existing tickets".
However thats not the case, I checked all tickets 3x, none is assigned to that account I wish to delete. Also no closed tickets (which should be disregarded for this task).

**What did you expect to happen**:
I would like to delete a user account without being stopped, and any tickets assigned to that person should become unassigned automatically.

**How to reproduce it (as minimally and precisely as possible)**:
Create dummy account, assign a ticket once, than assign same ticket to someone else, now try to delete the dummy account. Won't work.

**Anything else we need to know?**:

**Environment**:
- Trudesk Version: 1.0.9
- OS (e.g. from /etc/os-release): Windows 2019
- Node.JS Version: 11.12.0
- MongoDB Version: 4.0.6
- Is this hosted on cloud.trudesk.io: Nope

**Comment by polonel** [comment]
I can't reproduce using your reproduce method. I created an account and assigned it to a ticket. I think assigned another user and while the other user was selected I was able to delete the user I created.

If the user has ever created a ticket or commented on a ticket that the user will become disabled and not deleted. This is because the ticket/comment is linked to that user account for lookup. (To pull user information displayed for that ticket/comment) Even if you delete the ticket, the ticket goes into a soft delete state that is able to be restored for the Backup/Restore page. 

If the user also has active conversations on either themselves or the receiving conversation account. It will prevent from fully deleting because these conversations are linked to the user.

The other option was to mark the deleted user with a common "This User Has been Removed" account that will take the place where ever a deleted user information should display. I just found that has hard to determine who did what in an audit trail if it was just some "Deleted User".

I have considered not displaying the deleted account on the accounts page but figured it was easier to reinstate the account if needed.

Here is the check to disable to delete:
https://github.com/polonel/trudesk/blob/master/src/controllers/api/v1/users.js#L601
